### ISSUE TITLE
diff different day returns 0

### ISSUE THREAD

**Issue by Nomia** [moment/moment#3845]
**Description of the Issue and Steps to Reproduce:**
in the chrome devTool console
```
new Date(1489529963 * 1000)
Wed Mar 15 2017 06:19:23 GMT+0800 (CST)

new Date(1489491712 * 1000)
Tue Mar 14 2017 19:41:52 GMT+0800 (CST)
```

Obviously, these two are not the same day, but when I use the `diff` method of moment object, it returns `0`
```
moment(1489529963 * 1000).diff( moment(1489491712 * 1000), 'days' ); // => 0
```

it should return `1`

**Environment:**
Mac Sierra 10.12 Beta
Chrome DevTool / NodeJs

**ScreenShot**
![image](https://cloud.githubusercontent.com/assets/1707818/24068793/c46399ea-0bd2-11e7-97dd-9f1fc3fc7440.png)

**Comment by Nomia** [comment]
moment version: 2.17.0

**Comment by Nomia** [comment]
Another problem:
2017-02-28 is the last day of Feburary, the code below should return -1. But it returns -4 instead
![image](https://cloud.githubusercontent.com/assets/1707818/24068824/9080d1b4-0bd3-11e7-8ec3-1201196578e7.png)

**Comment by Nomia** [comment]
what is this?
![image](https://cloud.githubusercontent.com/assets/1707818/24068917/85a9a8d6-0bd5-11e7-92aa-96a4955c889a.png)

It not the desired behavior of `substract` method nor the `diff` method above.

**Comment by marwahaha** [comment]
Hope this helps:
> Note: Because this mirrors the native Date parameters, months, hours, minutes, seconds, and milliseconds are all zero indexed. Years and days of the month are 1 indexed.
> This is often the cause of frustration, especially with months, so take note!

http://momentjs.com/docs/#/parsing/array/

So:
* `moment([2017,3,1])` is April 1st.
* `moment([2017,02,28])` is March 28th. (four days before April 1st)

**Comment by marwahaha** [comment]
For your first question: since there is less than a day between the two dates, there are `0 days` that have passed. Try a different unit of measurement:
```
moment(1489529963 * 1000).diff( moment(1489491712 * 1000), 'seconds') // 38251
moment(1489529963 * 1000).diff( moment(1489491712 * 1000), 'hours') // 10
```

If this answers your questions, please close this issue. Thanks!

**Comment by Nomia** [comment]
OK, thanks, I got your point.
But it seems that I haven't make myself clear.

I don't think that the user of momentjs will take that the unit `days` in `diff` method means 24 * 60 * 60 * 1000 milliseconds, otherwise, they could just use unit `hours`, and do a math(divide by 24) to get the "days". 24 hours doesn't mean a day, it's just the duration. The `diff` method should just diff the days between two dates ignoring the hours or seconds or ms, since the `days` is the unit. If the we want to know the hours diff, we could just set the unit to `hours`.

What's the point to know if a date is 24 hours later than another?

The method and its behavior is confusing, or the `unit` parameter is confusing, specifically the `days` unit.

Above just my suggestion. I solved my problem myself not momentjs.
